harness/determined/exec/worker_process_wrapper.py

```diff
@@ -26,8 +27,17 @@ def run_all(ts: List[threading.Thread]) -> None:
 
 
 def main() -> int:
-    rank = os.environ.get("HOROVOD_RANK")
-    proc = subprocess.Popen(sys.argv[1:], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    parser = argparse.ArgumentParser()
+    # Different launcher may use different environment variable names to indicate rank.
+    parser.add_argument("rank_var_name")
+    parser.add_argument("cmd")
+    parser.add_argument("cmd_args", nargs=argparse.REMAINDER)
```

Comment:
Can you actually make this just `cmd` with `nargs=argparse.REMAINDER`?

I think that is the solution to the thing I was complaining about before with you and @azhou-determined.  Compare Case A (which fails) to Case B (which does not):

```python
import argparse

# Case A
parser = argparse.ArgumentParser()
parser.add_argument("-x")
parser.add_argument("cmd")
parser.add_argument("cmd_args", nargs=argparse.REMAINDER)
argsin = "-x x a -- b -- c"
args = parser.parse_args(argsin.split())
exp = "a -- b -- c"
got = " ".join([args.cmd] + args.cmd_args)
if got != exp:
    print(f"case A: '{argsin}'\nexpected: '{exp}'\nbut got:  '{got}'\n")


# Case B
parser = argparse.ArgumentParser()
parser.add_argument("-x")
parser.add_argument("cmd", nargs=argparse.REMAINDER)
argsin = "-x x a -- b -- c"
args = parser.parse_args(argsin.split())
exp = "a -- b -- c"
got = " ".join(args.cmd)
if got != exp:
    print(f"case B: '{argsin}'\nexpected: '{exp}'\nbut got:  '{got}'\n")
```